products/marketing_analytics/backend/hogql_queries/constants.py

```diff
@@ -145,3 +154,130 @@
+
+    # Determine kind and isIncreaseBad based on column type
+    kind = "unit"  # Default
+    is_increase_bad = False  # Default
+
+    # Check if it's a base column
+    for base_column in MarketingAnalyticsBaseColumns:
+        if key == base_column.value:
+            kind = COLUMN_KIND_MAPPING.get(base_column, "unit")
+            is_increase_bad = IS_INCREASE_BAD_MAPPING.get(base_column, False)
+            break
+    else:
+        # Check if it's a conversion goal or cost per conversion
+        if key.startswith(MarketingAnalyticsHelperForColumnNames.COST_PER):
+            kind = "currency"
+            is_increase_bad = True  # Cost per conversion - higher is bad
+        else:
+            # Regular conversion goal
+            kind = "unit"
+            is_increase_bad = False  # More conversions is good
+
+    # For string columns (Campaign, Source), preserve the string values
+    if kind == "unit" and key in [
+        MarketingAnalyticsBaseColumns.CAMPAIGN.value,
+        MarketingAnalyticsBaseColumns.SOURCE.value,
+    ]:
+        # String columns - no numeric processing needed
+        pass
+    else:
+        # For numeric columns, try to convert strings to numbers
+        if isinstance(value, str):
+            try:
+                value = float(value) if "." in value else int(value)
+            except (ValueError, TypeError):
+                value = None
+        if isinstance(previous, str):
+            try:
+                previous = float(previous) if "." in previous else int(previous)
+            except (ValueError, TypeError):
+                previous = None
+
+    # Handle percentage conversion for CTR
+    if kind == "percentage":
+        # CTR is already calculated as percentage in the query (multiplied by 100)
+        # No additional conversion needed
+        pass
+
+    # Calculate change percentage (only for numeric values)
+    change_from_previous_pct = None
+    if (
+        value is not None
+        and previous is not None
+        and isinstance(value, int | float)
+        and isinstance(previous, int | float)
+    ):
+        try:
+            if previous == 0:
+                # Handle special cases when previous is 0
+                if value == 0:
+                    # Both are 0: no change
+                    change_from_previous_pct = 0
+                elif value > 0:
+                    # From 0 to positive: use special large number to represent infinite growth
+                    change_from_previous_pct = int(InfinityValue.NUMBER_999999)
+                else:
+                    # From 0 to negative: use special large negative number to represent infinite decrease
+                    change_from_previous_pct = int(InfinityValue.NUMBER__999999)
+            else:
+                # Normal case: previous != 0
+                change_from_previous_pct = round(100 * (value - previous) / previous)
+        except (ValueError, ZeroDivisionError):
+            pass
```

Comment:
Same as the other PR, I don't think we do those 'Infinity' differences, just consider the actual general improvement.